src/posts/en/temporal-dead-zone-tdz-demystified.md

```diff
@@ -0,0 +1,193 @@
+
+# Well, TDZ is not quite so simple
+
+Taking a second look at the previous example, one could deduce that `let`/`const` declarations simply do not hoist, and that would explain the `ReferenceError`, right? **Nope**, that's an incorrect over-simplification *(and beware of uninformed resources claiming that!)*.
+
+Let's go back to the very first example in this article, replacing `var` with `let` and see what happens:
+
+```javascript
+let x = 'outer scope';
+(function() {
+    console.log(x);
+    let x = 'inner scope';
+}());
+```
+
+Can you guess what `console.log(x)` will print now? Well, actually, the answer is nothing -- the code above will throw a `ReferenceError` due to the TDZ semantics. That is because **`let`/`const` declarations do hoist**, but **they throw errors when accessed before being initialized** (instead of returning `undefined` as `var` would). I know the previous statement has already been expressed in this article, but this is actually the main point of TDZ and it is worth repeating as much as necessary (in fact, go ahead and do some memory training -- repeat the bold parts in this paragraph until it digs deep into your brain `;)`).
+
+Of course, this is still a bit of an over-simplification, I've put a lot of effort in balancing accuracy with simplicity to make an easy to remember and understand overview. Now onto the details...
+
+# The gory details
+
+Curious, are we? Let's travel a bit deeper into the TDZ then.
+
+The ECMAScript 2015 spec. clearly explains the `let`/`const` declarations hoisting and TDZ semantics in a [non-normative note](https://people.mozilla.org/~jorendorff/es6-draft.html#sec-let-and-const-declarations):
+
+> #### 13.2.1 Let and Const Declarations
+> NOTE `let` and `const` declarations define variables that are scoped to [the running execution context](https://people.mozilla.org/~jorendorff/es6-draft.html#sec-execution-contexts)’s [LexicalEnvironment](https://people.mozilla.org/~jorendorff/es6-draft.html#sec-execution-contexts). The variables are created when their containing [Lexical Environment](https://people.mozilla.org/~jorendorff/es6-draft.html#sec-lexical-environments) is instantiated but may not be accessed in any way until the variable’s *LexicalBinding* is evaluated. A variable defined by a *LexicalBinding* with an *Initializer* is assigned the value of its *Initializer*’s *AssignmentExpression* when the *LexicalBinding* is evaluated, not when the variable is created. If a *LexicalBinding* in a `let` declaration does not have an *Initializer* the variable is assigned the value `undefined` when the *LexicalBinding* is evaluated.
+
+Just in case your ECMAScript-fu is not sharp enough, I'll ~~dumb down~~ translate the relevant spec. parts to English:
```

Comment:
I would drop the `~~dumb down~~` part, only because most people that don't read/spend time with/understand the spec already feel "dumb" when they are confronted with spec material and often it puts them on the extreme defensive. This advice is purely a "sensitivity to the readers" recommendation, and I don't think you ever intended to be mean :)